pkg/kwok/server/patching_pod.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+
+	"sigs.k8s.io/kwok/pkg/log"
+)
+
+var (
+    universalDeserializer = serializer.NewCodecFactory(runtime.NewScheme()).UniversalDeserializer()
+)
+
+const (
+	KwokNodeLabel = "kwok.x-k8s.io/node"
```

Comment:
https://github.com/kubernetes-sigs/kwok/issues/1401#issuecomment-2892688170

As I mentioned, this isn't required.

This is just one security measure in the basic functionality demo. It's used to prevent new users from accidentally doing this in a production environment and causing it to crash!

Reply:
I made it optional, so patching pod handler is only installed when flags.Options.ManageNodesWithLabelSelector is specified, and we use the value of the flag as node label